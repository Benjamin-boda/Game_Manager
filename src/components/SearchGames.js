import React, {useState, useEffect, Fragment} from "react";
import SearchGamesDropDown, {SearchGamesDropDownNoItem} from "./SearchGamesDropDown";

const SearchGames = () => {
    const [data, setData] = useState([])
    const [dataIsReady, setDataIsReady] = useState(false)
    const [keyword, setKeyword] = useState("")
    
    
    useEffect(() => {
        async function getChickenApi() {
            if (keyword !== "") {
                try {
                    const response = await fetch(`https://chicken-coop.p.rapidapi.com/games?title=${keyword.toLowerCase()}`, {
                        "method": "GET",
                        "headers": {
                            "x-rapidapi-host": "chicken-coop.p.rapidapi.com",
                            "x-rapidapi-key": process.env.REACT_APP_SECRET_KEY
                        }
                    })
                    const json = await response.json()
                    const jsonPlat = json.result.map(({title, platform}) => platform)
                    const platformArray = jsonPlat.map((jsonPla) => {
                      if (jsonPla === "PS4") {
                        return "playstation-4"
                      } else if (jsonPla === "Switch") {
                        return "switch"
                      } else if (jsonPla === "X360") {
                        return "xbox-360"
                      } else if (jsonPla === "PS3") {
                        return "playstation-3"
                      } else if (jsonPla === "WIIU") {
                        return "wii-u"
                      } else if (jsonPla === "WII") {
                        return "wii"
                      } else if (jsonPla === "XONE") {
                        return "xbox-one"
                      } else if (jsonPla === "PC") {
                        return "pc"
                      } else if (jsonPla === "iOS") {
                        return "ios"
                      } else if (jsonPla === "3DS") {
                        return "3ds"
                      }
                    })
                    const jsonRandomGames = json.result.map((json, index) => {
                      return {"title":json.title, "platform": platformArray[index]}
                    })
                    setData(jsonRandomGames)
                    setDataIsReady(true)
                } catch(e) {
                    console.error(e)
                }
            }
        }
        getChickenApi()
    }, [keyword])

    const setKeywordInInput = (e) => {
        setKeyword(e.target.value)
    }

    return (
        <div>
            <input
                className="search"
                type="text"
                placeholder="Search a game..."
                onChange={setKeywordInInput}
                value={keyword}
            />
            <Fragment>
                <div>
                  {dataIsReady ? 
                    <ul className="search__dropdown">
                    {data.map((dat) => 
                      <SearchGamesDropDown key={dat.title + dat.platform} data={dat} setData={setData} setDataIsReady={setDataIsReady} />)}
                    </ul> 
                      : undefined}
                </div>
            </Fragment>
            
        </div>
    )
}



export default SearchGames;