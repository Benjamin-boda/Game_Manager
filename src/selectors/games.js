import moment from "moment";

//Get visible games

const getVisibleGames = (games, filters) => {
    return games.filter((game) => {
        const consoleMatch = game.platform.toLowerCase().includes(filters.console.toLowerCase());
        const textMatch = game.title.toLowerCase().includes(filters.text.toLowerCase());
        const statusMatch = game.played.toLowerCase().includes(filters.status.toLowerCase());

        return textMatch && statusMatch && consoleMatch;
    })
    .sort((a, b) => {
        if (filters.sortBy === "newDate") {
            return a.boughtAt < b.boughtAt ? 1 : -1;
        } else if (filters.sortBy === "oldDate") {
            return a.boughtAt < b.boughtAt ? -1 : 1;
        };
    })
};

export default getVisibleGames;