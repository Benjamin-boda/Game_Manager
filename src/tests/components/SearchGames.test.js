import { mount, shallow } from "enzyme";
import SearchGames from "../../components/SearchGames";
import games from "../fixtures/games";
import { Provider, useSelector, useDispatch } from "react-redux";
import React from "react";
import thunk from 'redux-thunk'
import { createMemoryHistory } from 'history'
import configureMockStore from "redux-mock-store";
import { shallowToJson } from "enzyme-to-json";

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);
const store = mockStore({});
const history = createMemoryHistory()

test("should render SearchGames page correctly", () => {
    const wrapper = shallowToJson(mount(
        <Provider store={store}>
            
                <SearchGames
                /> 
            
        </Provider>
        ))
    expect(wrapper).toMatchSnapshot();
});