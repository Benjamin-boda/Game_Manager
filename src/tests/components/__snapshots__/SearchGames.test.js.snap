// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchGames page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SearchGames>
    <div>
      <input
        onChange={[Function]}
        placeholder="Search a game..."
        type="text"
        value=""
      />
      <div>
        <ul>
          <SearchGamesDropDownNoItem>
            <li>
              <span>
                No result
              </span>
            </li>
          </SearchGamesDropDownNoItem>
        </ul>
      </div>
    </div>
  </SearchGames>
</Provider>
`;
